lib/MahoCLI/Commands/SysEncryptionKeyRegenerate.php

```diff
@@ -0,0 +1,226 @@
+
+        $select = $readConnection->select()
+            ->from($table)
+            ->where('value IS NOT NULL AND path IN (?)', $encryptedPaths);
+        $encryptedData = $readConnection->fetchAll($select);
+
+        if (empty($encryptedData)) {
+            $output->writeln('<info>No encrypted configurations to re-encrypt.</info>');
+            return Command::SUCCESS;
+        }
+
+        $outputTable = new Table($output);
+        $outputTable->setHeaders(['config_id', 'scope', 'scope_id', 'path']);
+        foreach ($encryptedData as $encryptedDataRow) {
+            $writeConnection->update(
+                $table,
+                ['value' => $this->encrypt($this->decrypt($encryptedDataRow['value']))],
+                ['config_id = ?' => $encryptedDataRow['config_id']],
+            );
+
+            $outputTable->addRow([
+                $encryptedDataRow['config_id'],
+                $encryptedDataRow['scope'],
+                $encryptedDataRow['scope_id'],
+                $encryptedDataRow['path'],
+            ]);
+        }
+
+        $output->writeln('<info>The following configurations were just re-encrypted, make sure to test all of them.</info>');
+        $outputTable->render();
+
+        Mage::dispatchEvent('encryption_key_regenerated', [
+            'output' => $output,
+            'encrypt_callback' => [$this, 'encrypt'],
+            'decrypt_callback' => [$this, 'decrypt'],
+        ]);
+
+        if (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
+            $output->writeln('');
+            $output->writeln('<error>Warning: phpseclib/mcrypt_compat is installed. This package can cause encryption issues and should be removed.</error>');
+            $output->writeln('<error>Please remove it using: composer remove phpseclib/mcrypt_compat</error>');
+            $output->writeln('');
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function decrypt(#[\SensitiveParameter] string $data): string
+    {
+        if ($this->isOldEncryptionKeyM1 && function_exists('mcrypt_module_open')) {
+            $key = $this->oldEncryptionKey;
+            $handler = mcrypt_module_open(MCRYPT_BLOWFISH, '', MCRYPT_MODE_ECB, ''); // @phpstan-ignore constant.notFound,constant.notFound
+            $initVector = mcrypt_create_iv(mcrypt_enc_get_iv_size($handler), MCRYPT_RAND); // @phpstan-ignore function.notFound,function.notFound,constant.notFound
+            mcrypt_generic_init($handler, $key, $initVector); // @phpstan-ignore function.notFound
+            $data = mdecrypt_generic($handler, (string) base64_decode((string) $data)); // @phpstan-ignore function.notFound
+            mcrypt_generic_deinit($handler); // @phpstan-ignore function.notFound
+            mcrypt_module_close($handler); // @phpstan-ignore function.notFound
+            return str_replace("\x0", '', trim($data));
+        }
+
+        $decoded = (string) base64_decode($data);
```

Comment:
I think we should use `sodium_base642bin` here too (and inverse in this->encrypt()), but we would need to handle exceptions.

Reply:
right, pushed it now, for the exception I think I still return '', interrupting the flow there will end up in half-converted data and unrecoverable data so... I think it's probably better to continue

Reply:
The only other option I could think of is to try to decrypt / encrypt every value before writing anything to the database or updating local.xml. But that would require quite some refactoring.

Reply:
it would be better, we could think about a transaction...

Reply:
I'm still wondering if returning false might be better than `''` or throwing. Unsure though since that might have rippling changes.

Reply:
I don't know, the return '' but logging the exception seems a good compromise to me. if the crypted data is corrupted nothing will work anyway but breaking the flow worries me more

Reply:
Yeah, in the end I think it's fine how it is. Furthermore, I tested with the backend config and it seems you can't encrypt an empty string there. If you leave a field blank it's NULL in core_config_data.